File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    var mapp= {}
    for (let ele of s){
        if (mapp[ele]){
            mapp[ele]+=1
        }
        else{
            mapp[ele]=1
        }
    }
     console.log(mapp)
   for (const key of Object.keys(mapp)){
        if (mapp[key]===1){
            return s.indexOf(key)
        }
    }
    return -1
};

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        mapp=dict()
        for i in s:
            mapp[i]=mapp.get(i,0)+1
        # print(mapp)
        for i in mapp:
            if mapp[i]==1:
                return list(s).index(i)
        return -1
        